src/components/Reown/TransactionFeedbackModal.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect } from 'react';
+import $ from 'jquery';
+import { t } from 'ttag';
+import { useDispatch } from 'react-redux';
+import { types } from '../../actions';
+import ReactLoading from 'react-loading';
+import { colors } from '../../constants';
+
+export const MODAL_ID = 'transactionFeedbackModal';
+
+/**
+ * Component that shows a modal with feedback for transactions
+ * Shows loading, success or error message and provides retry option on failure
+ */
+export function TransactionFeedbackModal({ isError, isLoading = true, errorMessage, onClose, manageDomLifecycle }) {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    manageDomLifecycle(`#${MODAL_ID}`);
+  }, []);
+
+  const handleRetry = () => {
+    dispatch({ type: types.REOWN_SEND_TX_RETRY });
+    onClose();
+  };
+
+  const handleDismiss = () => {
+    if (isError) {
+      dispatch({ type: types.REOWN_SEND_TX_RETRY_DISMISS });
+    }
+    onClose();
+  };
+
+  const renderContent = () => {
+    if (isLoading) {
+      return (
+        <>
+          <div className="modal-body text-center">
+            <ReactLoading type="spin" color={colors.purpleHathor} height={32} width={32} className="d-inline-block" />
+            <p className="mt-3">{t`Processing transaction...`}</p>
+          </div>
+          <div className="modal-footer">
+            <button type="button" className="btn btn-secondary" onClick={handleDismiss}>
+              {t`Cancel`}
+            </button>
+          </div>
```

Comment:
Does it make sense to allow cancelling the request while it's processing?

Reply:
Added a notice making it clear that cancelling doesn't actually cancel the transaction and only closes the modal.

What do you think? 

I can also remove the button

Reply:
I think it's ok with the note but I'd rather keep the modal opened so we guarantee to the user that the tx was processed. What happens if the processing fails? Will the user be notified after the modal is closed?

Reply:
Refactored this feedback modal to use the FeedbackModal component, which will not show the cancel button here